pkg/urbit/vere/saga.c

```diff
@@ -209,6 +209,65 @@ _cmp_epocs(const void* lef_v, const void* rih_v)
   return len_i == ren_i ? strcmp(lef_c, rih_c) : len_i - ren_i;
 }
 
+static c3_o
+_check_epoc_sanity(      u3_epoc* poc_u,
+                         c3_path* pax_u,
+                   const c3_c*    ent_c,
+                         c3_d*    nex_d,
+                         c3_l*    mex_l)
+{
+  //  check epoch's event range for contiguousness with earlier epochs.
+  //  first epoch may start at arbitrary event nr, so we don't check it.
+  c3_d fir_d = u3_epoc_first_commit(poc_u);
+  if ( (0 != *nex_d) && (*nex_d != fir_d) ) {
+    u3l_log("saga: snapshot for epoch %s not contiguous: "
+            "expected first event of %" PRIu64 ", got %" PRIu64 "\n",
+            u3_epoc_path_str(poc_u), *nex_d, fir_d);
+    return c3n;
+  }
+  *nex_d = u3_epoc_last_commit(poc_u) + 1;
+
+  //  load the snapshot so we can validate its mug
+  if ( 0 != *mex_l ) {
+    //NOTE  careful, will clobber all existing noun references!
+    c3_path_push(pax_u, ent_c);
+    //TODO  want _boot_from_epoc_snapshot but it ends up crashing in events.c
+    u3m_init();
+    u3e_load(c3_path_str(pax_u));
+    u3m_pave(c3n);
```

Comment:
The issue with `_boot_from_epoc_snapshot()` is that it assumes that `u3m_init()` has been called already (this assumption is undocumented, but I'll explicitly add it to the documentation). To resolve the crash in `events.c`, I think that the following change to `_cw_sane()` will work:
```c
u3m_init();
u3e_live(u3_Host.dir_c);
// ...
u3_saga* const log_u = u3_saga_open(/* snip */);
```